boolean/boolean.py

```diff
@@ -1032,55 +1094,61 @@ def start_operation(ast, operation):
                 ast[0].append(ast[1](*ast[2:], eval=eval))
                 ast = ast[0]
 
-    expr = expr.replace(" ", "")
-    length = len(expr)
+    if isinstance(expr, str):
+        tokenized = tokenizer(expr)
+    else:
+        try:
+            tokenized = list(expr)
+        except:
```

Comment:
I don't like much the idea of a raw `except:`... Also, why not simply just let the `for` below complain, and just do nothing here if it's not a string... converting to `list` would make it load everything to memory and `expr` could be a big generator. I'd propose, instead, checking to see if `expr` is a callable, and call it here, it might return a generator or an iterate.